src/VisualStudio/Core/Impl/SolutionExplorer/SymbolTree/RootSymbolTreeItemSourceProvider.cs

```diff
@@ -211,5 +213,24 @@ void OnItemPropertyChanged(object sender, PropertyChangedEventArgs e)
                 }
             }
         }
+
+        void AddToDictionary(string currentFilePath, RootSymbolTreeItemCollectionSource source)
+        {
+            if (!_filePathToCollectionSources.TryGetValue(currentFilePath, out var sources))
+            {
+                sources = [];
+                _filePathToCollectionSources[currentFilePath] = sources;
+            }
+
+            sources.Add(source);
+        }
+
+        void RemoveFromDictionary(string currentFilePath, RootSymbolTreeItemCollectionSource source)
+        {
+            if (_filePathToCollectionSources.TryGetValue(currentFilePath, out var sources))
+            {
+                sources.Remove(source);
```

Comment:
we should probably remove the key as well if there are no more sources.  Otherwise on renames we're going to be leaving old file paths around in the dictionary.

Looks like the multidictionary previously removed the key as well - https://github.com/icsharpcode/ILSpy/blob/master/ICSharpCode.Decompiler/Util/MultiDictionary.cs#L59